src/error.rs

```diff
@@ -0,0 +1,61 @@
+use std::error;
+use std::fmt;
+/// The error type returned by methods in this crate.
+pub enum Error<E> {
```

Comment:
Maybe this could be easier to do just by using `thiserror` crate?

Reply:
Nice that worked great